src/app/loadout-builder/process.ts

```diff
@@ -365,19 +404,33 @@ function byStatMix(lockedItems: readonly LockedItemType[] | undefined, assumeMas
     }
   }
 
+  // Handle armour 2.0 mods
+  if (lockedArmor2Mods) {
+    for (const lockedMod of lockedArmor2Mods) {
+      for (const stat of lockedMod.mod.investmentStats) {
+        lockedModStats[stat.statTypeHash] = lockedModStats[stat.statTypeHash] || 0;
+        lockedModStats[stat.statTypeHash] += stat.value;
+      }
+    }
+  }
+
   return (item: DimItem): string[] => {
     const stats = item.stats;
 
     if (!stats || stats.length < 3) {
-      return emptyStats;
+      return [JSON.stringify({ stats: emptyStats })];
```

Comment:
I don't think there's any need to generate this every time, when we know it's always going to be an identical string.